tests/test_fields.py

```diff
@@ -0,0 +1,430 @@
+"""
+Based on Django's field tests:
+https://github.com/django/django/tree/stable/3.0.x/tests/forms_tests/field_tests
+"""
+import datetime
+import logging
+from enum import Enum
+
+from django.conf import settings
+from django.core.validators import EMPTY_VALUES
+from django.forms import ValidationError, fields
+from django.test import SimpleTestCase
+from django_api_forms import (AnyField, BooleanField, DictionaryField, EnumField, FieldList, Form, FormField,
+                              FormFieldList, RequestValidationError)
+
+
+def log_input(val):
+    """
+    Logs info about attempted form field input values.
+
+    Mainly for tests that can raise ValidationError, because ValidationError
+    sometimes doesn't show the attempted value and a raised exception doesn't
+    return value for assert (which does show the attempted value).
+
+    Also helpful for showing attempted input with assertRaisesMessage where
+    the input value is defined in a loop.
+    """
+    logging.info('attempted input: "%s", type: "%s"', val, type(val))
+
+
+class BooleanFieldTests(SimpleTestCase):
+    FALSEY_VALUES = [
+        False,
+        0,
+        '0',
+        'False',
+        'false',
+        #  'FaLsE',
```

Comment:
Looks like Django's BooleanField handles this differently: https://github.com/django/django/blob/master/tests/forms_tests/field_tests/test_booleanfield.py#L40

Would it be possible to get rid of the custom BooleanField and use the one from django?

Reply:
Hi, I created my own BooleanField because I didn't liked the behaviour of the Django's BooleanFIeld. The main reason is, that I had issues with explicit true / false values. Django variation is created only to handle `checked` stated (like HTML checkbox). From my point of view, in APIs you need much better flexibility for controling boolean input. For example to allow `None` value if required or to specify explicit false (`False` != `None`).

Reply:
It might be a good idea to add a comment to BooleanField and the docs about this.

DRF implemented their own BooleanField and NullBooleanField as well: https://github.com/encode/django-rest-framework/blob/master/rest_framework/fields.py#L697-L790

Looks like they're explicitly checking for even more values, like "Y" and "N".

Reply:
I already referenced this problem in issue #10. Django also implemented similar [NullBooleanField](https://docs.djangoproject.com/en/3.0/ref/models/fields/#nullbooleanfield) as DRF. 

I am still unsure how to handle different "non JSON native" values for boolean. In one hant, it's pretty cool to have flexibility like this, on the other hand, it's pretty weird to accept values like this, because it's not a real boolean. From my point of view, it's a value error. 

What do you think @pawl?

Reply:
I guess DRF is accepting "non JSON native" values for boolean because they also process the "html/js" Content-type.

>it's pretty weird to accept values like this, because it's not a real boolean

Agreed. It might make sense for django_api_forms to check for only bool types and error on strings (like a json schema validator) since it's only accepting JSON and msgpack (both of which have native values for boolean).